### PR TITLE
Jt orders stuff

### PR DESCRIPTION
**Pull request by Joshua-Tukman**
Functionality for user stories 34 and 35

### CODE REVIEW

File: app/models/merchant.rb [hunk 1]
```diff
@@ -25,4 +26,13 @@ def distinct_cities
     item_orders.distinct.joins(:order).pluck(:city)
   end
 
+  def pending_items
+    @items = Item.where(merchant_id: id).joins(:orders).where(:orders => {:status => 'pending'})
+  end
+
+  def pending_orders
```
Comment (mikez321) [inline app/models/merchant.rb:33]: Ahhh nice here

File: db/seeds.rb [hunk 2]
```diff
@@ -24,14 +25,14 @@
                     password_confirmation: 'admin',
                     role: 2)
 
-merchant = User.create(name: 'Merchant1',
+merchant = bike_shop.users.create(name: 'Merchant1',
```
Comment (kmcgrevey) [inline db/seeds.rb:28]: now that relationship exists for the role:1 users . . .

File: app/models/order.rb [hunk 3]
```diff
@@ -35,4 +35,13 @@ def fulfill_item(item)
       self.status = 0
     end
   end
+
+  def merchant_item_quantity(merchant_id)
+    items.where(merchant_id: merchant_id).sum('quantity')
+  end
+
+  def merchant_item_subtotal(merchant_id)
+    subtotal = items.where(merchant_id: merchant_id).group(:id).sum('items.price * quantity')
```
Comment (mikez321) [inline app/models/order.rb:44]: We could clean these up in a refactor by putting the .'s in a line

File: spec/models/merchant_spec.rb [hunk 4]
```diff
@@ -11,54 +11,88 @@
 
   describe "relationships" do
     it {should have_many :items}
+    it {should have_many :users}
   end
 
   describe 'instance methods' do
```
Comment (kmcgrevey) [inline spec/models/merchant_spec.rb:17]: why is this all turned off?

File: app/models/user.rb [hunk 5]
```diff
@@ -2,8 +2,9 @@ class User < ApplicationRecord
   validates :email, uniqueness: true, presence: true
   has_secure_password
   validates_presence_of :name, :address, :city, :state, :zip, :password_digest
-  has_many :orders
 
+  has_many :orders
+  belongs_to :merchants, optional:true
```
Comment (mikez321) [inline app/models/user.rb:7]: This optional:true piece is sweet!

File: spec/models/user_spec.rb [hunk 6]
```diff
@@ -15,6 +15,7 @@
 
   describe "relationships" do
     it {should have_many :orders}
+    #it {should belong_to :merchant}
```
Comment (mikez321) [inline spec/models/user_spec.rb:18]: Should this be commented out?

**Comment by mikez321** [comment]
Looks good, like a lot of models tests as well!  Gotta love 'em

**Pull request merged**